store/store_test.go

```diff
@@ -145,6 +147,127 @@ func TestStore_Append_BadHeader(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestStore_Append(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	t.Cleanup(cancel)
+
+	suite := headertest.NewTestSuite(t)
+
+	ds := sync.MutexWrap(datastore.NewMapDatastore())
+	store := NewTestStore(t, ctx, ds, suite.Head(), WithWriteBatchSize(4))
+
+	head, err := store.Head(ctx)
+	require.NoError(t, err)
+	assert.Equal(t, head.Hash(), suite.Head().Hash())
+
+	const workers = 10
+	const chunk = 5
+	headers := suite.GenDummyHeaders(workers * chunk)
+
+	errCh := make(chan error, workers)
+	var wg stdsync.WaitGroup
+	wg.Add(workers)
+
+	for i := range workers {
+		go func() {
+			defer wg.Done()
+			// make every append happened in random order.
+			time.Sleep(time.Duration(rand.Intn(10)) * time.Millisecond)
+
+			err := store.Append(ctx, headers[i*chunk:(i+1)*chunk]...)
+			errCh <- err
+		}()
+	}
+
+	wg.Wait()
+	close(errCh)
+	for err := range errCh {
+		assert.NoError(t, err)
+	}
+
+	// wait for batch to be written.
+	time.Sleep(100 * time.Millisecond)
+
+	// assert.Eventually(t, func() bool {
+	head, err = store.Head(ctx)
+	assert.NoError(t, err)
+	assert.Equal(t, int(head.Height()), int(headers[len(headers)-1].Height()))
+
+	// 	return int(head.Height()) == int(headers[len(headers)-1].Height())
+	// }, time.Second, time.Millisecond)
+	assert.Equal(t, head.Hash(), headers[len(headers)-1].Hash())
+}
+
+func TestStore_Append_stableHeadWhenGaps(t *testing.T) {
```

Comment:
In sync aggreed to add to the test subscription to gaps and seeing that heightSub can handle them. Spoiler: it should break